src/Modules/Admin/UsersList/UsersList.tsx

```diff
@@ -1,7 +1,175 @@
-import React from 'react'
+import {
+  Box,
+  Stack,
+  Typography,
+  Table,
+  TableBody,
+  TableCell,
+  tableCellClasses,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Skeleton,
+  Pagination,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { Visibility } from '@mui/icons-material';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { axiosInstanceAdmin, USERS_URL } from '../../../Services/END_POINTS/ADMIN/URLS';
 
 export default function UsersList() {
+  const navigate = useNavigate();
+
+  interface User {
+    _id: string;
+    userName: string;
+    email: string;
+    phoneNumber: number;
+    country: string;
+    createdAt: number;
+    role: string;
+    profileImage: string;
+  }
+
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(0);
+  const pageSize = 10;
+  const getUsersList = async (pageSize: number, currentPage: number) => {
+    try {
+      const res = await axiosInstanceAdmin.get(USERS_URL.GET_ALL_USERS, { params: { size: pageSize, page: currentPage }});
+      setUsers(res.data.data.users);
+      const totalCount = res?.data?.data?.totalCount || 0;
+      setTotalPages(Math.ceil(totalCount / pageSize));
+    } catch (error) {
+      console.error('Error fetching users:', error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getUsersList(pageSize, currentPage);
+  }, []);
+  const handlePageChange = async (event: React.ChangeEvent<unknown>, page: number) => {
+    setCurrentPage(page);
+    await getUsersList(pageSize, page);
+   
+  };
+  const StyledTableCell = styled(TableCell)(() => ({
+    [`&.${tableCellClasses.head}`]: {
+      backgroundColor: 'rgba(226, 229, 235, 1)',
+      color: 'rgba(30, 38, 62, 1)',
+    },
+    [`&.${tableCellClasses.body}`]: {
+      fontSize: 14,
+      color: '#333',
+    },
+  }));
+
+  const StyledTableRow = styled(TableRow)(() => ({
+    '&:nth-of-type(odd)': {
+      backgroundColor: 'rgba(248, 249, 251, 1)',
+    },
+    '&:nth-of-type(even)': {
+      backgroundColor: '#ffffff',
+    },
+    '&:last-child td, &:last-child th': {
+      border: 0,
+    },
+  }));
+
+  const formatDate = (timestamp: number) => {
+    const date = new Date(timestamp);
+    return `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
+  };
+
+
+  const renderSkeletonRows = () => (
```

Comment:
Since we are using this component in multiple places, we can extract it into a reusable one that will take the number of rows and coloumns and render them correspondingly.